spec/lib/ddq-backend-mysql.spec.js

```diff
@@ -0,0 +1,382 @@
+                });
+            });
+
+            expect(() => {
+                instance.connect(() => {});
+            }).toThrow();
+        });
+        // This test is a bit redundant, as lib/configValidation should throw an
+        // error if a connection option is falsy. This provides branch coverage,
+        // though.
+        it("doesn't utilize connection options if they are falsy", () => {
+            instance.config.database = null;
+            instance.connect(() => {});
+        });
+    });
+    describe(".disconnect", () => {
+        it("ends the connection", () => {
+            instance.disconnect(() => {});
+            expect(instance.connection.end).toHaveBeenCalledWith(jasmine.any(Function));
+        });
+    });
+    describe(".listen", () => {
+        beforeEach(() => {
+            instance.currentlyPolling = false;
+        });
+        it("sets the flags and calls functions", () => {
+            timersMock.setTimeout.andReturn(true);
+            expect(instance.poller).toBe(null);
+            expect(instance.restorer).toBe(null);
+            instance.listen(() => {});
+            expect(instance.currentlyPolling).toBe(true);
+            expect(instance.poller).not.toEqual(false);
+            expect(instance.restorer).not.toEqual(false);
+            expect(timersMock.setTimeout).toHaveBeenCalled();
+        });
+    });
+    describe(".pausePolling", () => {
+        it("clears the timeouts and sets the flags", () => {
+            instance.poller = true;
+            instance.restorer = true;
+            instance.currentlyPolling = true;
+            instance.listen(() => {});
+            instance.pausePolling();
+            expect(instance.poller).toBe(null);
+            expect(instance.restorer).toBe(null);
+            expect(instance.currentlyPolling).toBe(false);
+            expect(timersMock.clearTimeout).toHaveBeenCalled();
+        });
+    });
+    describe(".poll", () => {
+        beforeEach(() => {
+            timersMock.setTimeout.andCallFake((callback) => {
+                if (timersMock.setTimeout.callCount < 2) {
+                    callback();
+                } else {
+                    return;
+                }
+            });
+            instance.connection.query.andCallFake((query, queryOptions, callback) => {
+                callback();
+            });
+        });
+        it("calls .findData", () => {
+            instance.listen(() => {});
+            expect(instance.connection.query).toHaveBeenCalled();
+        });
+    });
+    describe(".resumePolling", () => {
+        beforeEach(() => {
+            instance.currentlyPolling = false;
+        });
+        it("sets the flag and calls functions", () => {
+            timersMock.setTimeout.andReturn(true);
+            expect(instance.poller).toBe(null);
+            expect(instance.restorer).toBe(null);
+            instance.resumePolling();
+            expect(instance.currentlyPolling).toBe(true);
```

Comment:
I don't think that it's written quite enough to warrant its own function. If it were easier to write the test to include an option that would alter the expectation for falsy values then it might be warranted, but jasmine's syntax doesn't make that very easy.